File: backend/config/load_env.py
import os
from pathlib import Path
from dotenv import load_dotenv

BASE_DIR = Path(__file__).resolve().parent.parent

node_env = os.getenv("NODE_ENV", "development")

# In production (Docker), env variables are injected by docker-compose env_file
if node_env == "production":
    print(f"Running in production mode - using environment variables from Docker")
else:
    # For local development, load from .env.local file
    env_file = BASE_DIR / ".env.local"
    
    if env_file.exists():
        load_dotenv(env_file)
        print(f"Loaded environment from: {env_file.name} (NODE_ENV: {node_env})")
    else:
        print(f"⚠️  Warning: Could not load {env_file.name} from {env_file}")

File: backend/config/load-env.js
const path = require('path');

function loadGlobalEnv(dotenv) {
  const nodeEnv = process.env.NODE_ENV || 'development';
  
  // In production (Docker), env variables are injected by docker-compose env_file
  if (nodeEnv === 'production') {
    console.log(`Running in production mode - using environment variables from Docker`);
    return { parsed: process.env };
  }

  // For local development, load from .env.local file
  const envFile = '.env.local';
  const envPath = path.resolve(__dirname, '..', envFile);

  const result = dotenv.config({ path: envPath });

  if (result.error) {
    console.warn(`⚠️  Warning: Could not load ${envFile} from ${envPath}`);
  } else {
    console.log(`Loaded environment from: ${envFile} (NODE_ENV: ${nodeEnv})`);
  }
  return result;
}

module.exports = { loadGlobalEnv };

